Add OrgOwner roles to users when they are made the owner of a newly created Organization

philipr-za opened the pull request:
Addresses issue https://github.com/big-neon/bn-api/issues/64

- Check that a role is not a duplicate when adding it to a user
- Add the OrgOwner role when an Org is created with that user as the owner and when the org is updated.

Timo614 commented on api/src/controllers/organizations.rs:
```diff
@@ -107,7 +115,12 @@ pub fn update_owner(
+                Ok(updated_organization) => {
+                    let _ = DbUser::find(&updated_organization.owner_user_id, &*connection)
+                        .unwrap()
+                        .add_role(Roles::OrgOwner, &*connection);
```
We should probably check to see if the old organization owner is still the owner of any other organizations after the update here and then remove the OrgOwner role if not.

May be worth handling the result from the `DbUser#find` method above -- although there should never be a case where the uuid doesn't exist given the foreign key constraints on the db so probably not super critical and none of the other errors it can return seem that likely given its just a find (https://docs.diesel.rs/diesel/result/enum.Error.html https://docs.diesel.rs/diesel/result/enum.DatabaseErrorKind.html).

philipr-za commented:
So what I ended up doing was moving the OrgOwner logic into the DB layer. 
- When a Org is created the user who is the owner will get the OrgOwner role.
- Added a remove_role method on User
- The set_owner method will now give the new owner the OrgOwner role and check if the old owner owns any other Orgs and if not remove their OrgOwner role.
- Updated the tests for User and Organization
- Added the change_owner test

mikethetike commented on db/src/models/users.rs:
```diff
@@ -114,6 +116,20 @@ impl User {
+        DatabaseError::wrap(
+            ErrorCode::UpdateError,
+            "Could not remove role from user",
+            diesel::update(self)
```
This `diesel::update` should be refactored into a common function to reduce duplication between `add_role` and `remove_role`

philipr-za commented:
Thanks for the input, the Rustacean way is becoming clearer every day. Addressed all the comments.
